### DISCUSSION TITLE
Help to bring up fonoster docker environment

### DISCUSSION THREAD

**Discussion by txrubel** [fonoster/yaps#240] [Q&A] [answered]
Hello,

This is really an awesome project. I have built this fonoster docker image and running this in my laptop. Everything looks good. But now I am a bit stuck on, how should I use the fonoster cli tool for my own environment.

I want to create some project, provider and number for my own environment as describe [here](https://learn.fonoster.com/docs/getting_started).

Could you please help me on this? is there any doc on it?

Thanks

**Comment by psanders** [comment, marked as answer]
Did you use the [docker-in-docker](https://github.com/fonoster/fonoster/blob/main/docs/operator/deploy-with-docker.md) installer? 

The important thing is to identify the correct endpoint information and disable TLS on the client-side(by default TLS is disabled on the server). To connect to the local instance you need to:

First, find your credentials at `$(pwd)/fonoster/config`. It will look like this:

```bash
{
   "accessKeyId":"fonoster",
   "accessKeySecret":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
}
```

Next, you must disable TLS on the client-side by setting the environment variable `ALLOW_INSECURE` to true.

Finally, connect your CLI using the following `localhost:50051` as the endpoint.

**Reply by txrubel** [reply]
Dear @psanders, Thanks a lot for your quick response

yeah I am using the [docker-in-docker](https://github.com/fonoster/fonoster/blob/main/docs/operator/deploy-with-docker.md) installation.

Found all credentials. I will check them with AS_ALLOW_INSECURE and let you know all updates.

**Reply by psanders** [reply]
I see how that might be confusing (we definitely need to add a guide for this). Since your Fonoster instance has no TLS endpoint you need to disable TLS on the client too. In Linux/Osx you will do something like this:

```bash
ALLOW_INSECURE=true fonoster auth:login 
```

Or 

```bash
export ALLOW_INSECURE=true
fonoster auth:login
```

**Reply by txrubel** [reply]
Thanks and Appreciate @psanders ,
Now I can logged in, 
![image](https://user-images.githubusercontent.com/38957915/142656224-cdcd8d10-dbee-4af6-9952-e8616e6e3244.png)

**Reply by txrubel** [reply]
Just wanted to share we needed to use credentials in file config/user_credentials

**Comment by psanders** [comment]
Wao! I love it!

**Reply by ghost** [reply]
Hey,

I want to be able to "call my web app" comprising STT and TTS services and Fonoster seems to be built for exactly that (and Asterisk is giving my a hard time). For regulatory reasons, all services must run on premise.

So I installed Fonoster on Docker locally (on macOS) as described in [docker/README.md](https://github.com/fonoster/fonoster/blob/main/docker/README.md) -- so using defaults only.
That seems to have worked and ended with the repeating log line `🔔 [INFO]: Waiting for SIP Proxy to be ready... 🕐` (I guess that's normal without further configuration).

Then, I tried logging in (from outside of Docker) as suggested above - expect using

1. localhost instead of the Docker host IP (because the container's port is exposed) and
2. port 51051 instead 50051 because the default seems to have changed since the original question.

However, I receive the following error

```
Access your Fonoster infrastructure
Press ^C at any time to quit.
? api endpoint localhost:51051
? access key id fonoster
? access key token *********************************************************************
? ready? Yes
13 INTERNAL: Received RST_STREAM with code 0
Accessing endpoint localhost:51051... Invalid credentials or endpoint
(node:60261) Error Plugin: @fonoster/ctl: minTimeout is greater than maxTimeout
module: @oclif/config@1.18.16
task: runHook postrun
plugin: @fonoster/ctl
root: /opt/homebrew/lib/node_modules/@fonoster/ctl
See more details with DEBUG=*
(Use `node --trace-warnings ...` to show where the warning was created)
```

@psanders Did I do anything wrong or miss anything? All the documentation I could find so far assumes using Fonoster as SaaS. Is there a specific guide than the one I mentioned above for setting up Fonoster Community Edition locally?

Thanks in advance and for your work on an open-source Twilio alternative! 💪 

PS: @txrubel Your link in the original post is no longer valid but I guess it is now https://learn.fonoster.com/docs/getting_started/overview.
